app/modules/code_provider/code_provider_controller.py

```
@@ -241,6 +241,14 @@ async def get_branch_list(
         from app.modules.utils.logger import setup_logger
 
         logger = setup_logger(__name__)
+
+        provider_type = os.getenv("CODE_PROVIDER", "github").lower()
+        if provider_type == "github" and not os.path.isdir(repo_name):
+            try:
+                GithubService._validate_repo_name_format(repo_name)
+            except ValueError as e:
+                raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)